src/libs/ajax/file-browser-providers/AzureBlobStorageFileBrowserProvider.ts

```diff
@@ -206,6 +206,37 @@ const AzureBlobStorageFileBrowserProvider = ({
         method: 'DELETE',
       });
     },
+    moveFile: async (sourcePath: string, destinationPath: string): Promise<void> => {
+      const {
+        sas: { url: originalSasUrl },
+      } = await storageDetailsPromise;
+
+      const sourceBlobUrl = new URL(originalSasUrl);
+      sourceBlobUrl.pathname += `/${sourcePath}`;
+
+      const destinationBlobUrl = new URL(originalSasUrl);
+      destinationBlobUrl.pathname += `/${destinationPath}`;
+
+      await fetchOk(destinationBlobUrl.href, {
+        method: 'PUT',
+        headers: {
+          'x-ms-copy-source': sourceBlobUrl.href,
+        },
+      });
+
+      // Note
```

Comment:
Discussion in Slack: https://broadinstitute.slack.com/archives/C0DSD41QT/p1690402807787449